server/src/test/java/io/crate/integrationtests/DiskThresholdDeciderIT.java

```diff
@@ -0,0 +1,365 @@
+            for (ShardRouting shard : indexShardRoutingTable.shards()) {
+                assertThat(shard.state()).isEqualTo(ShardRoutingState.STARTED);
+                if (shard.currentNodeId().equals(nodeId)) {
+                    shardRoutings.add(shard);
+                }
+            }
+        }
+        return shardRoutings;
+    }
+
+    /**
+     * Index documents until all the shards are at least WATERMARK_BYTES in size, and return the size of the smallest shard
+     * @param indexName
+     */
+    private long createReasonableSizedShards(String indexName) throws InterruptedException, ExecutionException {
+        // Table created by createIndex() takes default schema.
+        // UseRandomizedSchema is not involved when creating index directly.
+        String tableName = "doc." + indexName;
+        while (true) {
+
+            execute("insert into " + tableName + "(x) values (?)", new Object[]{randomAlphaOfLengthBetween(1000, 2000)});
+
+            ensureGreen();
+            execute("optimize table " + tableName + " with (max_num_segments=1)");
+
+            refresh();
+
+            var indicesStats = client().admin().indices().stats(new IndicesStatsRequest().indices(indexName).store(true)).get();
+            final ShardStats[] shardStatses = indicesStats.getIndex(indexName).getShards();
+
+            final long[] shardSizes = new long[shardStatses.length];
+            for (ShardStats shardStats : shardStatses) {
+                shardSizes[shardStats.getShardRouting().id()] = shardStats.getStats().getStore().sizeInBytes();
+            }
+
+            final long minShardSize = Arrays.stream(shardSizes).min().orElseThrow(() -> new AssertionError("no shards"));
+            if (minShardSize > WATERMARK_BYTES) {
+                return minShardSize;
+            }
+        }
+    }
+
+    private void refreshDiskUsage() throws ExecutionException, InterruptedException {
+        ((InternalClusterInfoService) internalCluster().getMasterNodeInstance(ClusterInfoService.class)).refresh();
+        // if the nodes were all under the low watermark already (but unbalanced) then a change in the disk usage doesn't trigger a reroute
+        // even though it's now possible to achieve better balance, so we have to do an explicit reroute. TODO fix this?
+        final ClusterInfo clusterInfo = internalCluster().getMasterNodeInstance(ClusterInfoService.class).getClusterInfo();
+        if (StreamSupport.stream(clusterInfo.getNodeMostAvailableDiskUsages().values().spliterator(), false)
+            .allMatch(cur -> cur.value.getFreeBytes() > WATERMARK_BYTES)) {
+
+            var clusterRerouteResponse = client().admin().cluster()
+                .execute(ClusterRerouteAction.INSTANCE, new ClusterRerouteRequest()).get();
+            assertAcked(clusterRerouteResponse);
+        }
+
+        var clusterHealthRequest = new ClusterHealthRequest()
```

Comment:
`prepareHealth` is removed in https://github.com/crate/crate/commit/b6fede9b6ad58781f501f41d2251e9f8eb811bf6 so using request directly